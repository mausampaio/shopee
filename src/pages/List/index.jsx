import { useContext, useEffect } from "react";
import { ProductCard } from "../../components/ProductCard";
import { ListContent } from "./styles";
import { Container } from "../../components/Container";
import { Store } from "../../context";
import { useQuery } from "../../hooks/useQuery";
import useWindowDimensions from "../../hooks/useWindowDimensions";
import { Loading } from "../../components/Loading";
import SearchForm from "../../components/SearchForm";

const List = () => {
  const context = useContext(Store);
  const query = useQuery();
  const { width } = useWindowDimensions();
  const searchValue = query.get("search")?.trim();

  useEffect(() => {
    searchValue
      ? context.getSearchedProduct(searchValue)
      : context.getProducts();
  }, [searchValue]);

  return context.loading ? (
    <Loading />
  ) : (
    <Container>
      {searchValue ? (
        <h1>Resultados para "{searchValue}"</h1>
      ) : (
        <h1>Bem-vindo!</h1>
      )}
      {width <= 575 ? <SearchForm /> : null}
      <ListContent>
        {context.products.length ? (
          context.products.map((product) => (
            <ProductCard productDetails={product} key={product.id} />
          ))
        ) : (
          <h2>Nenhum resultado encontrado para esta busca</h2>
        )}
      </ListContent>
    </Container>
  );
};

export default List;
