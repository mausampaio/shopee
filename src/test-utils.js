import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

export const mockProduct = {
  id: 1,
  name: "Produto de teste",
  price: 1.0,
  description: "Produto gerado para testes",
  url: "produto-de-teste",
  image: "./produto-teste.jpg",
};

export const mockCart = {
  id: mockProduct.id,
  name: mockProduct.name,
  price: mockProduct.price,
  quantity: 10,
};

export const mockContext = {
  cart: [],
  products: [],
  product: {},
  loading: false,
  getCartTotal: () => 0,
};

const RouterProvider = ({ children }) => {
  return <BrowserRouter>{children}</BrowserRouter>;
};

const customRender = (ui, options) => {
  return render(<RouterProvider>{ui}</RouterProvider>, { ...options });
};

export * from "@testing-library/react";

export { customRender as render };
