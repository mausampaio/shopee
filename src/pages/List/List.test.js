import List from ".";
import { Store } from "../../context";
import { mockContext, mockProduct, render } from "../../test-utils";

const ContextProvider = ({ children }) => (
  <Store.Provider
    value={{
      ...mockContext,
      products: [mockProduct],
      getProducts: () => {
        return;
      },
    }}
  >
    {children}
  </Store.Provider>
);

test("should render products from the context", () => {
  const { getByText } = render(<List />, {
    wrapper: ContextProvider,
  });
  expect(getByText(mockProduct.name)).toBeInTheDocument();
});

test("should match snapshot", () => {
  const { asFragment } = render(<List />, {
    wrapper: ContextProvider,
  });
  expect(asFragment()).toMatchSnapshot();
});
