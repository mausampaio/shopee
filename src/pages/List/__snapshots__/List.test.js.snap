// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  .c8 {
  width: 100%;
  min-width: 44px;
  height: 44px;
  background-color: var(--secondary);
  border-radius: 4px;
  color: var(--primary);
  font-size: 18px;
  font-weight: 500;
  cursor: pointer;
}

.c7 {
  width: 226px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  background-color: var(--primary);
  width: 258px;
  padding: 16px;
  border-radius: 7px;
  box-shadow: 0 4px 8px rgba(0,0,0,0.1);
}

.c3 {
  width: 226px;
  height: 226px;
  overflow: hidden;
  background-color: var(--primary);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 16px;
}

.c3 img {
  max-width: 100%;
  max-height: 100%;
}

.c4 {
  font-size: 18px;
  line-height: 24px;
  font-weight: 500;
  display: block;
  margin-bottom: 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 32px;
  line-height: 24px;
  font-weight: 500;
  margin-bottom: 8px;
  display: block;
}

.c5 span {
  font-size: 14px;
}

.c6 {
  font-size: 12px;
  line-height: 24px;
  font-weight: 400;
  margin-bottom: 16px;
  display: block;
}

.c1 {
  display: grid;
  grid-template-columns: repeat(4,1fr);
  width: 100%;
}

.c1 > * {
  margin-bottom: 32px;
}

.c1 > h2 {
  grid-column: 1 / 5;
  text-align: center;
}

.c0 {
  width: 80%;
  max-width: 1280px;
  margin: 0 auto;
}

@media screen and (max-width:575px) {
  .c2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media screen and (max-width:575px) {
  .c3 {
    margin: 0;
  }
}

@media screen and (max-width:768px) {
  .c1 {
    grid-template-columns: repeat(2,1fr);
    justify-items: center;
  }

  .c1 > h2 {
    grid-column: 1 / 3;
  }
}

@media screen and (max-width:575px) {
  .c1 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media screen and (max-width:768px) {
  .c0 {
    width: 90%;
  }
}

<div
    class="c0"
  >
    <h1>
      Bem-vindo!
    </h1>
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <a
          href="/products/1"
        >
          <div
            class="c3"
          >
            <img
              alt="Produto de teste"
              src="./produto-teste.jpg"
            />
          </div>
        </a>
        <a
          class="c4"
          href="/products/1"
        >
          Produto de teste
        </a>
        <div
          class="c5"
        >
          <span
            style="font-size: 20px; line-height: 24px; font-weight: 500; padding-top: 0px;"
          >
            R$
          </span>
          <span
            style="font-size: 32px; line-height: 24px; font-weight: 500;"
          >
             1
          </span>
          <span
            style="font-size: 16px; line-height: 24px; font-weight: 500;"
          >
            ,00
          </span>
        </div>
        <span
          class="c6"
        >
          ou 10x de R$ 0,10
        </span>
        <div
          class="c7"
        >
          <button
            class="c8"
          >
            Comprar
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
